static/app/components/codeSnippet.tsx

```diff
@@ -48,71 +176,86 @@ export function CodeSnippet({
       : t('Unable to copy');
 
   return (
-    <Fragment>
-      {!hideActionBar && (
-        <CodeContainerActionBar>
-          {filename && <span>{filename}</span>}
-          <Tooltip delay={0} isHoverable={false} title={tooltipTitle} position="bottom">
-            <UnstyledButton
+    <Wrapper>
+      {filename && (
+        <Header>
+          {filename}
+          {!hideCopyButton && (
+            <CopyButton
               type="button"
+              size="xs"
+              borderless
+              isInHeader
               onClick={handleCopy}
+              title={tooltipTitle}
+              tooltipProps={{delay: 0, isHoverable: false, position: 'left'}}
               onMouseLeave={() => setTooltipState('copy')}
             >
-              <IconCopy />
-            </UnstyledButton>
-          </Tooltip>
-        </CodeContainerActionBar>
+              <IconCopy size="xs" />
+            </CopyButton>
+          )}
+        </Header>
       )}
```

Comment:
Copy button will always be visible (unless `hideCopyButton` is true) when the header is rendered:
<img width="630" alt="Screen Shot 2022-09-30 at 12 11 25 PM" src="https://user-images.githubusercontent.com/44172267/193340481-01dff548-5530-4668-949b-7b58de5a738a.png">